### ISSUE TITLE
Add inpaint not masked to controlnet as mask mode

### ISSUE THREAD

**Issue by ghpkishore** [continue-revolution/sd-webui-segment-anything#51]
Given that automatic1111 has mask mode of inpaint not masked, controlnet should also have that. Since segment anything has a controlnet option, there should be a mask mode to send to controlnet from SAM. This way I can mask a small part of the problem image which I do not want to be disturbed and change the rest of it with controlnet

**Comment by continue-revolution** [comment]
Sure. It will be supported after I get up

**Comment by Arron17** [comment]
> Given that automatic1111 has mask mode of inpaint not masked, controlnet should also have that. Since segment anything has a controlnet option, there should be a mask mode to send to controlnet from SAM. This way I can mask a small part of the problem image which I do not want to be disturbed and change the rest of it with controlnet

Whilst you're waiting for this repo to be updated you can do this now with the latest ControlNet Extension Update. It allows you to use the Inpainting Model without an image in the ControlNet Window, but in the InPainting Window.

https://github.com/Mikubill/sd-webui-controlnet/issues/968

So as a workaround at the moment, in Segment Everything, select the option to copy to inpaint upload, DO NOT tick the box for ControlNet. Then setup controlnet for inpainting, it will take the image from Segment Anything that's been copied to the Inpaint Upload Section (You won't see it but it's there) and you can use either Inpaint Masked or Inpaint not masked and both will work.

**Comment by continue-revolution** [comment]
Yes. In the next update #57 there will be a paint not mask option **only in the txt2img** panel. In the img2img panel, you should select inpaint not mask in the inpaint upload panel. I may need another day of two to figure out all potential problems of that update.
